runtime/onert/backend/trix/DevContext.cc

```diff
@@ -225,30 +238,57 @@ void DevContext::runOneBatch(uint32_t dev_num, ModelID model_id, input_buffers *
 
   const auto &dev_handle = _dev_handles.at(dev_num);
   int req_id;
-  if (createNPU_request(dev_handle, model_id_at_device, &req_id))
+
+  if (auto error_code = createNPU_request(dev_handle, model_id_at_device, &req_id))
   {
     throw std::runtime_error("Unable to create NPU request with model id (" +
-                             std::to_string(model_id_at_device) + ")");
+                             std::to_string(model_id_at_device) + ")" +
+                             " error code : " + std::to_string(error_code));
   }
 
-  if (setNPU_requestData(dev_handle, req_id, input_bufs, in_info, output_bufs, out_info))
+  if (auto error_code =
+        setNPU_requestData(dev_handle, req_id, input_bufs, in_info, output_bufs, out_info))
   {
     removeNPU_request(dev_handle, req_id);
     throw std::runtime_error("Unable to create NPU request for model id (" +
-                             std::to_string(model_id_at_device) + ")");
+                             std::to_string(model_id_at_device) + ")" +
+                             " error code : " + std::to_string(error_code));
+  }
+
+  // NOTE submitNPU_request is not thread-safe(?). It is rarely hanging(unresponsive).
+  //      Ultimately, to solve this problem, we have to either use other thread-safe API or
+  //      change submitNPU_request to be thread-safe, but both works take time.
+  //      As a workaround, let's allow hanging thread.
+  // TODO Change submitNPU_request to be thread-safe or replaced with other thread-safe API
+  std::packaged_task<int(npudev_h, int)> task(submitNPU_request);
```

Comment:
Q) this approach incurs two threads per 1 batch, which results in 2X number of threads compared to previous approach ? Then, it may affects performance ?

Reply:
Yes. The codes here are not intended to improve performance. They can slightly weaken performance. The codes here are just to run `submutNPU_request` on a new thread, not the current thread because of avoiding hanging on main thread.

Reply:
Ok. Then, Let's check significant difference on performance.

Reply:
There is a difference, 10% ~ 20%.
It's not difference on slightly weaken performance :(
```bash
# BACKENDS=trix /usr/bin/nnfw-test/Product/out/unittest/nnpackage_run /opt/workspace/manual/mv1.q8 --shape_prepare="[0,[10,224,224,3]]" --load:raw /opt/workspace/manual/mv1.q8/input_all.tv2b --output_sizes="[0,20020]" --dump:raw /opt/workspace/manual/mv1.q8/result_parallel.tv2b -w 2 -r 100
Package Filename /opt/workspace/manual/mv1.q8
/opt/workspace/manual/mv1.q8/result_parallel.tv2b.0 is generated.
===================================
MODEL_LOAD   takes 1.540 ms
PREPARE      takes 56.435 ms
EXECUTE      takes 35.038 ms
- MEAN     :  35.038 ms
- MAX      :  63.506 ms
- MIN      :  25.003 ms
- GEOMEAN  :  34.330 ms
===================================
```